core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -249,4 +257,67 @@ long getMaxNumberOfThreads() {
 
     }
 
+    static class MaxMapCountCheck implements Check {
+
+        private final long limit = 1 << 18;
```

Comment:
I think there is confusion about how this stuff works, please, start up elasticsearch and look. try different heap sizes. there is more or less a fixed cost, you know mapping in all the files the jvm needs, libraries all this. on my machine its around 500.

but from there on, its all about file handles. and its a fraction of the number of file handles.